### PR TITLE
356 picard2.18

### LINKED ISSUE #356: picard Markduplicate reports inacurate optical duplicate
We need to update to the newer version of Picard as it seems a lot more accurate especially when the PCR duplicate rate is high.

### PR DESCRIPTION
**Pull request by lukbut**
An updated version of Picard (v2.18.23) has been released and needs to be incorporated in the data processing pipeline - closes #356

### CODE REVIEW

**Comment by lukbut** [comment]
Resolves #371

**Comment by lukbut** [comment]
@tcezard The requested changes have now been implemented, you can have another look if you wish.

File: tests/assets/test_crawlers/test_picard2_markdup.metrics [hunk 1]
```diff
@@ -1,9 +1,112 @@
...
 
 ## METRICS CLASS	picard.sam.DuplicationMetrics
-LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
-	0	0	0	0	0	0	?	
-Unknown Library	4509	536418	5703	799	90990	90990	0.169657	
+LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	SECONDARY_OR_SUPPLEMENTARY_RDS	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
+Unknown Library	4945	527174	3953	5169	775	79100	57983	0.150077
+
+## HISTOGRAM	java.lang.Double
+BIN	VALUE
```
Comment (mwhamgenomics) [inline tests/assets/test_crawlers/test_picard2_markdup.metrics:11]: Do we need all of this test data if the parser `break`s when it reaches here?

File: tests/assets/test_crawlers/test_picard_markdup.metrics [hunk 2]
```diff
@@ -1,9 +1,112 @@
...
 
 ## METRICS CLASS	picard.sam.DuplicationMetrics
-LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
-	0	0	0	0	0	0	?	
-Unknown Library	4426	637017	5616	866	114983	114974	0.180555	15140319228
+LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	SECONDARY_OR_SUPPLEMENTARY_RDS	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
+Unknown Library	4945	527174	3953	5169	775	79100	57983	0.150077	5054793
+
+## HISTOGRAM	java.lang.Double
+BIN	VALUE
```
Comment (mwhamgenomics) [inline tests/assets/test_crawlers/test_picard_markdup.metrics:11]: Same as above

File: tests/assets/test_crawlers/test_run_dir/10015AT/10015AT0001/a_run_id_1_S0_L001_markdup.metrics [hunk 3]
```diff
@@ -1,9 +1,112 @@
...
 
 ## METRICS CLASS	picard.sam.DuplicationMetrics
-LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
-	0	0	0	0	0	0	?	
-Unknown Library	4426	637017	5616	866	114983	114974	0.180555	15140319228
+LIBRARY	UNPAIRED_READS_EXAMINED	READ_PAIRS_EXAMINED	SECONDARY_OR_SUPPLEMENTARY_RDS	UNMAPPED_READS	UNPAIRED_READ_DUPLICATES	READ_PAIR_DUPLICATES	READ_PAIR_OPTICAL_DUPLICATES	PERCENT_DUPLICATION	ESTIMATED_LIBRARY_SIZE
+Unknown Library	4945	527174	3953	5169	775	79100	57983	0.150077	5054793
+
+## HISTOGRAM	java.lang.Double
+BIN	VALUE
```
Comment (mwhamgenomics) [inline tests/assets/test_crawlers/test_run_dir/10015AT/10015AT0001/a_run_id_1_S0_L001_markdup.metrics:11]: Same as above

**Review by mwhamgenomics — COMMENTED** [review]
Should we add Picard as a versioned tool as well?

File: tests/assets/test_crawlers/test_picard2_markdup.metrics (hunk 1 above)
Comment (lukbut) [reply tests/assets/test_crawlers/test_picard2_markdup.metrics:11]: I think there's value in retaining the original format of a test sample input file, leaving it as close to the generated one as possible, but I'll change it if @tcezard and you believe it's better that way.

File: tests/assets/test_crawlers/test_run_dir/10015AT/10015AT0001/a_run_id_1_S0_L001_markdup.metrics (hunk 3 above)
Comment (lukbut) [reply tests/assets/test_crawlers/test_run_dir/10015AT/10015AT0001/a_run_id_1_S0_L001_markdup.metrics:11]: Please refer to my comment above.

File: tests/assets/test_crawlers/test_picard_markdup.metrics (hunk 2 above)
Comment (lukbut) [reply tests/assets/test_crawlers/test_picard_markdup.metrics:11]: Please refer to my comment above.

**Comment by lukbut** [comment]
> Should we add Picard as a versioned tool as well?

@mwhamgenomics Where would that be, in Project Management?

**Review by tcezard — APPROVED** [review]
Please clip the metrics files in order to limit the number of lines added.
otherwise looks fine

**Review by tcezard — DISMISSED** [review]
I forgot that we said you should add picard in the toolset and add a command to extract the version

**Comment by lukbut** [comment]
@mwhamgenomics @tcezard I have now implemented the requested changes, kindly review.

**Review by tcezard — APPROVED** [review]

**Pull request merged**